pandas/io/parsers/base_parser.py

```diff
@@ -1228,6 +1231,15 @@ def _isindex(colspec):
                     colspec = orig_names[colspec]
                 if _isindex(colspec):
                     continue
+                elif dtype_backend == "pyarrow":
+                    import pyarrow as pa
+
+                    dtype = data_dict[colspec].dtype
+                    if isinstance(dtype, pd.ArrowDtype) and pa.types.is_timestamp(
```

Comment:
Might make sense to keep date types too i.e. `pa.types.is_date`

Reply:
Added